ts/Stock/StockToolsBindings.ts

```diff
@@ -385,27 +395,43 @@ bindingsUtils.updateHeight = function (
     e: PointerEvent,
     annotation: Annotation
 ): void {
-    annotation.update({
-        typeOptions: {
-            height: this.chart.pointer.getCoordinates(e).yAxis[0].value -
-                (annotation.options.typeOptions.points as any)[1].y
-        }
-    });
+    const coordsY =
+        this.utils.getAssignedAxis(this.chart.pointer.getCoordinates(e).yAxis);
+
+    if (coordsY) {
+        annotation.update({
+            typeOptions: {
+                height: coordsY.value -
+                    (annotation.options.typeOptions.points as any)[1].y
+            }
+        });
+    }
 };
 
 // @todo
 // Consider using getHoverData(), but always kdTree (columns?)
 bindingsUtils.attractToPoint = function (
     e: PointerEvent,
     chart: Chart
-): Highcharts.NavigationBindingsAttractionObject {
+): Highcharts.NavigationBindingsAttractionObject | void {
     var coords = chart.pointer.getCoordinates(e),
-        x = coords.xAxis[0].value,
-        y = coords.yAxis[0].value,
+        coordsX = chart.navigationBindings?.utils.getAssignedAxis(coords.xAxis),
+        coordsY = chart.navigationBindings?.utils.getAssignedAxis(coords.yAxis),
```

Comment:
Better do one check for `chart.navigationBindings` then assign coordsX and coordsY. Optional chaining is quite bulky when transpiled to ES5, and here it's twice over.